File: backend/internal/database/models/partners.go
package models

import "time"

type Partner struct {
	// Unique partner identifier
	ID uint64 `gorm:"primaryKey;autoIncrement" json:"id" example:"1"`
	// Partner legal name
	Name string `gorm:"type:varchar(255);not null" json:"name" example:"ACME Corp"`
	// Legal status
	LegalStatus *string `gorm:"type:varchar(255)" json:"legal_status,omitempty" example:"SAS"`
	// Mailing address
	Address *string `gorm:"type:text" json:"address,omitempty" example:"5 Avenue Anatole France, Paris"`
	// Contact email
	Email string `gorm:"type:varchar(255);not null" json:"email" format:"email" example:"partners@acme.tld"`
	// Contact phone
	Phone *string `gorm:"type:varchar(100)" json:"phone,omitempty" example:"+33 1 23 45 67 89"`
	// Creation timestamp (UTC)
	CreatedAt *time.Time `json:"created_at,omitempty" format:"date-time"`
	// Short description
	Description *string `gorm:"type:text" json:"description,omitempty" example:"Sponsor"`
	// Type of partnership
	PartnershipType *string   `gorm:"type:varchar(255)" json:"partnership_type,omitempty" example:"sponsor"`
	CreatedAtLocal  time.Time `gorm:"autoCreateTime" json:"-"`
	UpdatedAtLocal  time.Time `gorm:"autoUpdateTime" json:"-"`
}

File: web/src/lib/api/contracts/partners.ts
export type PartnerDTO = {
  id: number;
  name: string;
  legal_status?: string | null;
  address?: string | null;
  email: string;
  phone?: string | null;
  created_at?: string | null;
  description?: string | null;
  partnership_type?: string | null;
};

export type Partner = {
  id: number;
  name: string;
  legalStatus?: string | null;
  address?: string | null;
  email: string;
  phone?: string | null;
  createdAt?: Date | null;
  description?: string | null;
  partnershipType?: string | null;
};

export const mapPartner = (dto: PartnerDTO): Partner => ({
  id: dto.id,
  name: dto.name,
  legalStatus: dto.legal_status,
  address: dto.address,
  email: dto.email,
  phone: dto.phone,
  createdAt: dto.created_at ? new Date(dto.created_at) : null,
  description: dto.description,
  partnershipType: dto.partnership_type,
});
